tests/test_app.py

```diff
@@ -1,69 +1,57 @@
-from todo_app import app
-
-@pytest.fixture
-def client():
-    file_path = find_dotenv('.env.test')
-    load_dotenv(file_path, override=True)
-
-    test_app = app.create_app()
-
-    with test_app.test_client() as client:
-        yield client
-
-def test_index_page_loads(monkeypatch, client):
-    
-    # Given
-    monkeypatch.setattr(requests, 'get', get_lists_stub)
-
-    # When
-    response = client.get('/')
-
-    # Then
-    assert response.status_code == 200
-    assert 'Just another to-do app.' in response.data.decode()
-
-def test_index_page_displays_trello_data(monkeypatch, client):
-    
-    # Given
-    monkeypatch.setattr(requests, 'get', get_lists_stub)
-
-    # When
-    response = client.get('/')
-
-    # Then
-    assert 'Test card' in response.data.decode()
-
-class StubResponse():
-    def __init__(self, fake_response_data):
-        self.fake_response_data = fake_response_data
-
-    def json(self):
-        return self.fake_response_data
-
-def get_lists_stub(url, params):
-    test_board_id = os.environ.get('TRELLO_BOARD_ID')
-
-    print(url)
-    fake_response_data = None
-    if url == f'https://api.trello.com/1/boards/{test_board_id}/lists':
-        fake_response_data = [
-            {
-                'idBoard': '1234ABCD',
-                'id': '890xyz',
-                'name': 'To Do',
-                'cards': [
-                    {
-                        'id': '456', 
-                        'name': 'Test card', 
-                        'dateLastActivity': '2022-09-06T15:45:55.959Z',
-                        'idBoard': '1234ABCD',
-                        'idList': '890xyz',
-                    },
-                ],
-            }
-        ]
-    return StubResponse(fake_response_data)
\ No newline at end of file
+import os
+import pytest
+from dotenv import load_dotenv, find_dotenv
+import mongomock
+import pymongo
+from datetime import datetime
+
+from todo_app import app
+from todo_app.mongo_db_config import MongoDbConfig
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        test_app = app.create_app()
+        with test_app.test_client() as client:
+            yield client
+
+def test_index_page_loads(client):
+    # Given
+    config = MongoDbConfig()
+
+    collection = pymongo.MongoClient(
+        config.COSMOS_CONNECTION_STRING
+    )[config.COSMOS_DB_NAME]["to_dos"]
+    collection.insert_one(_get_db_document())
+
+    # When
+    response = client.get('/')
+
+    # Then
+    assert response.status_code == 200
+    assert 'Just another to-do app.' in response.data.decode()
+
+def test_index_page_displays_data(client):
+    # Given
+    config = MongoDbConfig()
+
+    collection = pymongo.MongoClient(
+        config.COSMOS_CONNECTION_STRING
+    )[config.COSMOS_DB_NAME]["to_dos"]
+    collection.insert_one(_get_db_document())

```

Comment:
Minor, but given this is quite a broad test already it feels reasonable that this would take advantage of existing functions within the app; would replacing this by the existing functionality simplify this? Is there any reason to prefer to do it fresh? (Not a rhetorical question, I often find I'm not totally sold on this approach!)
```suggestion
    db_repo = DatabaseRepository()
    db_repo.create_item_on_todo_list("Test card")
```

Reply:
Yeah, I take your point. I think my counterpoint would be that this test _shouldn't_ fail if someone hypothetically breaks the `create_item_on_todo_list` function, because in reality the page should still render independently of that function, so the test should mirror that?